crates/cheatcodes/spec/src/vm.rs

```diff
@@ -499,43 +499,57 @@ interface Vm {
     #[cheatcode(group = Evm, safety = Unsafe)]
     function readCallers() external returns (CallerMode callerMode, address msgSender, address txOrigin);
 
+    // -------- Gas Snapshots --------
+
+    /// Gets the gas used in the last call.
+    #[cheatcode(group = Evm, safety = Safe)]
+    function lastCallGas() external view returns (Gas memory gas);
+
+    /// Start a snapshot capture of the current gas usage.
+    #[cheatcode(group = Evm, safety = Unsafe)]
+    function startSnapshotGas(string calldata name) external returns (bool success);
```

Comment:
Consider also adding an arbitrary value snapshot

```
snapshotValue(string calldata name, uint256 value)
```

simple way to open up bunch more use cases:
- vm.snapshotValue(vm.lastCallGas)
- vm.snapshotValue(address(contract).code.length)

Reply:
yeah +1! I think snapshotting lastCallGas is the top way to get accurate snapshotting for an entire transaction (combined with using --isolate). So would love either a `lastCallGasSnapshot` or a custom snapshot like mark suggested!

Reply:
Supportive, added `snapshotValue` (incl. grouping)